portal/static/js/accountCreation.js

```diff
@@ -1,120 +1,114 @@
-            if (params.callback) {
+            var self = this;
+            if (!hasValue(params.apiUrl)) {
                 (params.callback).call(self, {"error": "API url is required."});
-            };
-            return false;
-        };
-        var errorMessage = "";
-        self.attempts++;
-        self.params = params;
-        $.ajax ({
-            type: (params.requestType ? params.requestType : "GET"),
-            url: String(params.apiUrl).replace("//", "/"),
-            contentType: "application/json; charset=utf-8",
-            dataType: "json",
-            async: params.sync ? false : true,
-            data: params.requestData
-        }).done(function(data) {
-            self.attempts = 0;
-            if (params.callback) {
-                (params.callback).call(self, {"data": data});
-            };
-        }).fail(function(xhr) {
-            if (self.attempts < self.maxAttempts) {
-                setTimeout ( function() { self.__request( self.params ); } , $.ajaxSetup().retryAfter );
-            } else {
-                var displayError = i18next.t("Server error occurred updating data.");
-                $("#error_response_text").html(displayError);
-                errorMessage = "Error processing request: " + params.apiUrl;
-                errorMessage += ";  response status code: " + (parseInt(xhr.status) === 0 ? "request timed out/network error": xhr.status);
-                errorMessage +=";  response text: " + (hasValue(xhr.responseText) ? xhr.responseText : "no response text returned from server");
-                tnthAjax.reportError(self.userId, "/patients/patient-profile-create", errorMessage, true);
+                return false;
+            }
+            var errorMessage = "";
+            self.attempts++;
+            self.params = params;
+            $.ajax ({
+                type: (params.requestType ? params.requestType : "GET"),
+                url: String(params.apiUrl).replace("//", "/"),
+                contentType: "application/json; charset=utf-8",
+                dataType: "json",
+                async: params.sync ? false : true,
+                data: params.requestData
+            }).done(function(data) {
                 self.attempts = 0;
-                if (params.callback) {
+                (params.callback).call(self, {"data": data});
+            }).fail(function(xhr) {
+                if (self.attempts < self.maxAttempts) {
+                    setTimeout ( function() { self.__request( self.params ); } , $.ajaxSetup().retryAfter );
+                } else {
+                    var displayError = i18next.t("Server error occurred updating data.");
+                    $("#error_response_text").html(displayError);
+                    errorMessage = "Error processing request: " + params.apiUrl;
+                    errorMessage += ";  response status code: " + (parseInt(xhr.status) === 0 ? "request timed out/network error": xhr.status);
```

Comment:
Nice error-handling :)
Would it be useful to generalize this for re-use in other AJAX requests?